lib/mayaHydra/flowViewportAPIExamples/footPrintNode/mhFootPrintNode.cpp

```diff
@@ -0,0 +1,666 @@
+    MObject currentMObj = thisMObject();
+    if (_oldMObject.isNull() || _oldMObject != currentMObj){
+        
+        _thisMObject = currentMObj;     
+        _oldMObject  = _thisMObject;
+
+        setupFlowViewportInterface();
+    }
+
+    if (plug == mWorldS) 
+    {
+        if (plug.isElement())
+        {
+            MArrayDataHandle outputArrayHandle = dataBlock.outputArrayValue( mWorldS );
+            outputArrayHandle.setAllClean();
+        }
+        dataBlock.setClean(plug);
+
+        return MS::kSuccess;
+    }
+    
+    return MS::kUnknownParameter;;
+}
+
+bool MhFootPrint::isBounded() const
+{
+    return true;
+}
+
+MBoundingBox MhFootPrint::boundingBox() const
+{
+    const double multiplier = _GetSizeInCentimeters();
+    return MBoundingBox( corner1 * multiplier, corner2 * multiplier);//corner1 and 2 are the bounding box corner of our geometry
+}
+
+void* MhFootPrint::creator()
+{
+    return new MhFootPrint();
+}
+
+//---------------------------------------------------------------------------
+//---------------------------------------------------------------------------
+// Plugin Registration
+//---------------------------------------------------------------------------
+//---------------------------------------------------------------------------
+
+//Macro to create input attribute for the maya node
+#define MAKE_INPUT(attr)	                    \
+    CHECK_MSTATUS(attr.setKeyable(true) );		\
+    CHECK_MSTATUS(attr.setStorable(true) );		\
+    CHECK_MSTATUS(attr.setReadable(true) );		\
+    CHECK_MSTATUS(attr.setWritable(true) );		\
+	CHECK_MSTATUS(attr.setAffectsAppearance(true) );
+
+//Macro to create output attribute for the maya node
+#define MAKE_OUTPUT(attr)			            \
+    CHECK_MSTATUS ( attr.setKeyable(false) );   \
+    CHECK_MSTATUS ( attr.setStorable(false) );	\
+    CHECK_MSTATUS ( attr.setReadable(true) );   \
+    CHECK_MSTATUS ( attr.setWritable(false) );
+
+MStatus MhFootPrint::initialize()
+{
+    MFnUnitAttribute    unitFn;
+    MFnNumericAttribute nAttr;
+
+    mSize = unitFn.create( "size", "sz", MFnUnitAttribute::kDistance);
+    MAKE_INPUT(unitFn);
+    CHECK_MSTATUS ( unitFn.setDefault(1.0) );
+
+    mWorldS = unitFn.create("worldS", "ws", MFnUnitAttribute::kDistance, 1.0);
+    unitFn.setWritable(true);
+    unitFn.setCached(false);
+    unitFn.setArray( true );
+    unitFn.setUsesArrayDataBuilder( true );
+    unitFn.setWorldSpace( true );
+
+    mColor = nAttr.create("color", "col", MFnNumericData::k3Double, 1.0);
+    MAKE_INPUT(nAttr);
+    CHECK_MSTATUS ( nAttr.setDefault(0.0, 0.0, 1.0) );
+
+    //Create dummy input attribute to trigger a call to the compute function on demand. as it's in the compute fonction that we add our scene indices
+    mDummyInput = nAttr.create("dummyInput", "dI", MFnNumericData::kInt, 1.0);
+    MAKE_INPUT(nAttr);
+    CHECK_MSTATUS ( nAttr.setDefault(1) );
```

Comment:
I question the need for this.  If the size and color attributes properly affect the output, we shouldn't need this dummy input.

Reply:
I integrated your changes where you removed the dummy attributes, thanks for your contribution.